File: 3194-minimum-average-of-smallest-and-largest-elements/3194-minimum-average-of-smallest-and-largest-elements.cpp
class Solution {
public:
    double minimumAverage(vector<int>& nums) {
        
        sort(nums.begin(),nums.end());
        vector<double>avgs;
        while(nums.size()>0){
            int n=nums.size();
            avgs.push_back((nums[0]+nums[nums.size()-1])/2.0);
            nums.erase(nums.begin()+0);
            nums.pop_back();
        }
        double ans=avgs[0];
        for(int i=0;i<avgs.size();i++) {
            if(ans>avgs[i]) ans=avgs[i];
        }
        return ans;
    }
};

File: 3194-minimum-average-of-smallest-and-largest-elements/3194-minimum-average-of-smallest-and-largest-elements.py
class Solution:
    def minimumAverage(self, nums: List[int]) -> float:
        a = len(nums)//2
        c = []
        for i in range(a):
            m = max(nums)
            n = min(nums)
            c.append((m+n)/2)
            nums.remove(m)
            nums.remove(n)
        return min(c)  